front/src/views/PlayoutHistory.vue

```diff
@@ -1,115 +1,121 @@
+                <option>100</option>
+                <option>1000</option>
+              </select>
+            </div>
           </div>
         </div>
-      </div>
-      <div class="field">
-        <label class="checkbox" for="chronological">
-          <input class="checkbox" type="checkbox" id="chronological" v-model="chronological" true-value="yes" false-value=""/>
-          Chronologically
-        </label>
-      </div>
-      <div class="field">
-        <button class="button is-primary">Search</button>
+        <div class="field is-align-self-flex-end">
+          <button :class="`button is-link ${isLoading ? 'is-loading' : ''}`">
+            Search
+          </button>
+        </div>
       </div>
     </form>
-    <div class="table-container">
-      <table class="table is-striped">
-        <thead>
-          <tr>
-            <th>Day</th>
-            <th>When</th>
-            <th>Artist</th>
-            <th>Title</th>
-            <th>Source</th>
-            <th>Source URI</th>
-          </tr>
-        </thead>
-        <tbody>
-          <tr v-for="result in results" :key="result.on_air">
-            <td>{{ new Date(result.on_air).toLocaleDateString() }}</td>
-            <td>{{ new Date(result.on_air).toLocaleTimeString() }}</td>
-            <td>{{ result.artist }}</td>
-            <td>{{ result.title }}</td>
-            <td class="is-size-7">{{ result.source }}</td>
-            <td class="is-size-7">{{ result.initial_uri }}</td>
-          </tr>
-        </tbody>
-      </table>
-    </div>
+    <CardList :results="results" :loading="isLoading" :error="isError" />
   </div>
 </template>
 
 <script>
-import http from '@/http'
+import http from "@/http";
+import { format } from "date-fns";
+import CardList from "../components/CardList.vue";
+import Datepicker from "vue3-datepicker";
 
 export default {
-  data () {
+  components: { CardList, Datepicker },
+  data() {
     return {
-      start: '',
-      end: '',
+      start: "",
+      end: "",
       limit: "10",
-      chronological: '',
-      results: []
-    }
-  },
-
-  computed: {
-
+      chronological: "",
+      results: null,
+      isLoading: true,
+      isError: false,
+    };
   },
-
   methods: {
-    getHistory () {
+    getHistory() {
+      this.isLoading = true;
+      this.isError = false;
       let params = new URLSearchParams();
       if (this.limit) {
         params.append("limit", this.limit);
       }
       if (this.start) {
-        params.append("start", this.start);
+        params.append("start", format(new Date(this.start), "yyyy-MM-dd"));
       }
       if (this.end) {
-        params.append("end", this.end);
-      }
-      if (this.chronological) {
-        params.append("chronological", this.chronological);
+        params.append("end", format(new Date(this.end), "yyyy-MM-dd"));
       }
-      http.get('/metadata_log', { params: params })
+      // if (this.chronological) {
+      params.append("chronological", "yes");
```

Comment:
I'd keep that checkbox, otherwise the default form would always show the first 10 tracks ever played. With the checkbox, unchecked by default, switching to this page shows directly the last 10 tracks.

The checkbox fits 2 use cases:
 - default form for a daily routine, checking what's been on air (although showing the last 10 is a bit short)
 - calendar search for a regular extract (spoiler: at some point it will be accompanied by a "download as CSV" button)

Reply:
Ok, I'll do it but maybe we can discuss this further. I cannot really see the meaning of seeing the first tracks of the day, instead it feels more natural to my opinion to always see the last ones

Reply:
Yes - that's the first case I listed, and in that case the browser shouldn't query with `chronological` at all.

The second case is for much larger extractions (days, months, likely a year), those are often more intuitive in chronological order.